src/components/molecules/Visualizer/hooks.ts

```diff
@@ -249,6 +259,97 @@ export default ({
     pageview(window.location.pathname);
   }, [isPublished, enableGA, trackingId]);
 
+  const [overriddenAlignSystem, setOverrideAlignSystem] = useState<WidgetAlignSystem | undefined>(
+    alignSystem,
+  );
+
+  const overrideWidgetPosition = useCallback((widgetId: string, options: SetUIPositionOptions) => {
+    if (
+      !widgetId ||
+      !["outer", "inner"].includes(options.zone) ||
+      !["left", "center", "right"].includes(options.section) ||
+      !["top", "middle", "bottom"].includes(options.area) ||
+      (options.section === "center" && options.area === "middle")
+    )
+      return;
+
+    let tarWidget: Widget | undefined;
+
+    setOverrideAlignSystem(alignSystem => {
+      if (!alignSystem) return alignSystem;
+      Object.keys(alignSystem).forEach(zoneName => {
+        const zone = alignSystem[zoneName as keyof WidgetAlignSystem];
+        if (zone) {
+          Object.keys(zone).forEach(sectionName => {
+            const section = zone[sectionName as keyof WidgetZone];
+            if (section) {
+              Object.keys(section).forEach(areaName => {
+                const area = section[areaName as keyof WidgetSection];
+                if (!tarWidget && area?.widgets) {
+                  let tarIndex;
+                  for (let i = 0, len = area.widgets.length; i < len; i += 1) {
+                    if (area.widgets[i].id === widgetId) {
+                      tarIndex = i;
+                      break;
+                    }
+                  }
+                  if (tarIndex !== undefined) {
+                    [tarWidget] = area.widgets.splice(tarIndex, 1);
+                  }
+                }
+              });
+            }
+          });
+        }
+      });
+      return { ...alignSystem };
+    });
+
+    setTimeout(() => {
```

Comment:
Here i separated the process of remove the target widget from current position and add it into the target positon. The adding is also using a `setTimeout` 0 to make it a async task. The reason for it is when do remove/add in sync and when the source & target area is the same one, the rest of widget inside this area also got reloaded. For example area is [A, B, C] when set widget C's position to first like [C, A, B], widget A & B will also reload. This behavior can be reproduced with a simplified react test project. But if make this process to two separated steps like remove first and then add it (two updates for react) A & B will not reload. I think this has something related with how react-dom update the real DOM and i didn't find a better solution for it.